src/main/java/com/google/devtools/build/lib/runtime/BlazeOptionHandler.java

```diff
@@ -361,6 +366,80 @@ DetailedExitCode parseOptions(List<String> args, ExtendedEventHandler eventHandl
     return DetailedExitCode.success();
   }
 
+  private void parseInvocationPolicyToRcNotesFile(InvocationPolicy invocationPolicy) {
+    List<AccumulatedFlagsFromInvocationPolicy> accumulator = new ArrayList<>();
+
+    List<String> commandNamesToParse = getCommandNamesToParse(commandAnnotation);
+
+    for (InvocationPolicyOuterClass.FlagPolicy flagPolicy :
```

Comment:
having a hard time following this loop. instead of effectively duplicating the invocation-policy enforcement code, could we tie the reporting into InvocationPolicyEnforcer#enforce?

Reply:
Done - this required a bit of restructuring to be able to preserve the command the flag applied to, but it's hopefully all wired up ok.